src/setup.cpp

```diff
@@ -834,3 +827,44 @@ void sound_init()
         break;
     }
 }
+
+void store_window_size()
+{
+    int new_width;
+    int new_height;
+    auto window = Draw.get_window();
+    SDL_GetWindowSize(window, &new_width, &new_height);
+    if (window_width != new_width && window_height != new_height)
+    {
+        // Need to store it
+        Config.push_config_state();
+        Config.set_config_file(kqres(eDirectories::SETTINGS_DIR, "kq.cfg").c_str());
+        Config.set_config_int(nullptr, "window_width", new_width);
+        Config.set_config_int(nullptr, "window_height", new_height);
+        Config.pop_config_state();
+    }
+}
+
+/*! \brief Ask for window mode.
+ * @returns the selected mode
+ */
+eDisplayMode prompt_display_mode()
+{
+    const char* options[] = { _("Full screen"), _("Window 1x"), _("Window 2x"), _("Window 3x"), _("Window 4x") };
+    int choice = Draw.prompt_ex(255, _("Select display mode"), options, 5);
```

Comment:
Do we know whether `Draw.prompt_ex()` will wrap to the bottom option if at the top element and UP is pressed (or wrap to the top option if at the bottom element and DOWN is pressed)?